test/Makefile.am

```diff
@@ -10,9 +10,9 @@ AM_CPPFLAGS	= -DTEST_SRCDIR=\"$(top_srcdir)/test\" \
 		  -I$(top_srcdir)/include \
 		  -I$(top_srcdir)/external/secp256k1/include
 
-noinst_LIBRARIES= libtest.a
+check_LTLIBRARIES = libtest.la
 
```

Comment:
I thought the `check_` prefix did not work with the `LIBRARIES` primary but I was mistaken. Reading the automake manual I now see that:

> The special prefix ‘check_’ indicates that the objects in question should not be built until the ‘make check’ command is run. Those objects are not installed either.
> 
> The current primary names are ‘PROGRAMS’, ‘LIBRARIES’, ‘LTLIBRARIES’, ‘LISP’, ‘PYTHON’, ‘JAVA’, ‘SCRIPTS’, ‘DATA’, ‘HEADERS’, ‘MANS’, and ‘TEXINFOS’.

Reply:
Also, `check_` prefix has slightly negative value versus `noinst_`, because it reduces the amount of code built by default, which prevents immediate notification of test breakage caused by a non-test code bug.

Reply:
Latest commit reverts `check_` prefix to `noinst_`